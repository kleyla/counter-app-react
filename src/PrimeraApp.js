import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitulo }) => {
  return (
    <div>
      <h1>{saludo}</h1>
      <h3>{subtitulo}</h3>
    </div>
  );
};

PrimeraApp.protoTypes = {
  saludo: PropTypes.string.isRequired,
};
PrimeraApp.defaultProps = {
  subtitulo: "Soy un subtitulo",
};

export default PrimeraApp;
