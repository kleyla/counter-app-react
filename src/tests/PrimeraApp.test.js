import React from "react";
// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

describe("Prueba en PrimeaApp", () => {
  //   test("Debe de mostrar el mensaje Hola, soy Karen", () => {
  //     const saludo = "Hola, soy Karen";
  //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });

  test("Debe de mostrar PrimeraApp correctamente", () => {
    const saludo = "Hola, soy Karen";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });
});
