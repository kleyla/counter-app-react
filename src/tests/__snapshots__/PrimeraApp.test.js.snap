// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en PrimeaApp Debe de mostrar PrimeraApp correctamente 1`] = `
<div>
  <h1>
    Hola, soy Karen
  </h1>
  <h3>
    Soy un subtitulo
  </h3>
</div>
`;
